biosteam/units/compressor.py

```diff
@@ -867,25 +867,40 @@ def _overwrite_subcomponent_id(self, subcomponent, i_stage):
         else:
             subcomponent.outs[0].ID = f"{ID}"
 
+    def reset_cache(self, **kwargs):
+        super().reset_cache(**kwargs)
+        if (self.pr is not None
```

Comment:
`reset_cache` will not work if `self.pr` is never set (e.g. if setup option 1 is chosen). Is this the intended behavior?

Reply:
Ahh, I misread the code earlier. Yeah, unit operations are not recreated. The `reset_cache` code did not work for both options, good catch.

I made some changes so that the main contribution of this PR are:
* Unit operations and streams are not renamed in option 1
* Allow for user to change pr, n_stages, the compressors, or the heat exchangers between simulations (and setup will rerun).

I'll go ahead and merge by tomorrow unless you found some other issues.

Thanks for reviewing the code!

Reply:
@yoelcortes I think we're good to merge. Thanks for the improvements 😃